src/commands/scp.ts

```diff
@@ -75,26 +75,22 @@ const scpAction = async (args: yargs.ArgumentsCamelCase<ScpCommandArgs>) => {
     throw "Could not determine host identifier from source or destination";
   }
 
-  const requestId = await provisionRequest(authn, args, host);
+  const { publicKey, privateKey } = await createKeyPair();
 
-  if (!requestId) {
+  const request = await provisionRequest(authn, args, host, publicKey);
```

Comment:
If the request is pre-existing then the returned request object from Datastore must have the same publicKey as the one we submitted from `createKeyPair`. That's going to be the case as long as our backend implementation is correct but we could also add a check here that compares the two public keys. wdyt?

Reply:
👍 , good idea. i added a small check here.